src/JaxkDev/DiscordBot/Bot/Client.php

```diff
@@ -187,73 +216,33 @@ public function getDiscordClient(): Discord{
 		return $this->client;
 	}
 
-	public function getPluginCommunicationHandler(): PluginCommunicationHandler{
-		return $this->pluginCommsHandler;
-	}
-	/*
-	 * Note, It will only show warning ONCE per channel/guild that fails.
-	 * Fix on the way hopefully.
-	 */
-	public function sendMessage(string $guild, string $channel, string $content): void{
-		if($this->thread->getStatus() !== Protocol::THREAD_STATUS_READY) return;
-
-		/** @noinspection PhpUnhandledExceptionInspection */
-		$this->client->guilds->fetch($guild)->done(function(Guild $guild) use($channel, $content) {
-			$guild->channels->fetch($channel)->done(function(Channel $channel) use($guild, $content) {
-				$channel->sendMessage($content);
-				MainLogger::getLogger()->debug("Sent message(".strlen($content).") to ({$guild->id}|{$channel->id})");
-			}, function() use($guild, $channel) {
-				MainLogger::getLogger()->warning("Failed to fetch channel {$channel} in guild {$guild->id} while attempting to send message.");
-			});
-		}, function() use($guild) {
-			MainLogger::getLogger()->warning("Failed to fetch guild ${guild} while attempting to send message.");
-		});
-	}
-
-	public function updatePresence(string $text, int $type): bool{
-		$presence = new Activity($this->client, [
-			'name' => $text,
-			'type' => $type
-		]);
-
-		try {
-			$this->client->updatePresence($presence);
-			return true;
-		} catch (Exception $e) {
-			return false;
-		}
+	public function getCommunicationHandler(): CommunicationHandler{
+		return $this->communicationHandler;
 	}
 
-	public function logDebugInfo(): void{
```

Comment:
In terms of debugging I think I need something to dump the entirety of the bot for example guilds, members, roles ID's and counts are semi useless for the now advanced API.

Reply:
d81479aa